docs: add docstring examples to four public functions

mdciofalo opened the pull request:
## Description
Closes #1082.

## Changes
- Added examples to `matrix_ops/calculate_vector_matrix_dimension`
- Added examples to `states/mutually_unbiased_basis`
- Added examples to `nonlocal_games/quantum_hedging` methods:
  - `max_prob_outcome_a_primal`
  - `max_prob_outcome_a_dual`
  - `min_prob_outcome_a_primal`
  - `min_prob_outcome_a_dual`
- Added examples to methods in `nonlocal_games/xor_game`:
  - `nonsignaling_value`

## Checklist
  -  [x] Use `ruff` for errors related to code style and formatting.
  -  [x] Verify all previous and newly added unit tests pass in `pytest`.
  -  [x] Check the documentation build does not lead to any failures. `Sphinx` build can be checked locally for any failures related to your PR
  -  [x] Use `linkcheck` to check for broken links in the documentation
  -  [x] Use `doctest` to verify the examples in the function docstrings work as expected.

mdciofalo commented:
This is ready for review. Let me know if anything needs to be changed!

purva-thakre commented on toqito/states/mutually_unbiased_basis.py:
```diff
@@ -26,6 +26,13 @@ def mutually_unbiased_basis(dim: int) -> list[np.ndarray]:
         M_1 = \left\{\frac{|0\rangle + |1\rangle}{\sqrt{2}}, \frac{|0\rangle - |1\rangle}{\sqrt{2}}\right\}
         M_2 = \left\{\frac{|0\rangle + i|1\rangle}{\sqrt{2}}, \frac{|0\rangle - i|1\rangle}{\sqrt{2}}\right\}
 
+    >>> from toqito.matrix_ops import mutually_unbiased_basis
```
```suggestion
    >>> from toqito.states import mutually_unbiased_basis
```
The import path was incorrect

purva-thakre commented:
@mdciofalo Formatting failure https://github.com/vprusso/toqito/actions/runs/14202026069/job/39793553173?pr=1099#step:3:37

BTW you can check these locally. Please check the contributing guide in the documentation. If the instructions there don't make sense, add a comment to #1095.

mdciofalo commented:
Hi! I've addressed the doctest failures and fixed the trailing whitespace that caused formatting issues. I also ran pytest, ruff, make html, make doctest, and linkcheck locally — all passed with no failures.

The Read the Docs build is failing due to cvxopt missing UMFPACK headers (umfpack.h). This appears unrelated to the changes in this PR, which only affect docstrings and pass all other tests locally.

purva-thakre commented:
@mdciofalo Something is wrong with RTD. We are seeing similar failures across multiple PRs.
